lib/immortal.rb

```diff
@@ -124,17 +124,23 @@ def destroy!
     end
 
     def destroy_without_callbacks
-      self.class.unscoped.update_all({ :deleted => true }, "id = #{self.id}")
+      self.class.unscoped.update_all({ deleted: true, updated_at: current_time_from_proper_timezone }, "id = #{self.id}")
       @destroyed = true
       reload
       freeze
     end
 
     def recover!
-      self.class.unscoped.update_all({ :deleted => false }, "id = #{self.id}")
+      self.class.unscoped.update_all({ deleted: false, updated_at: current_time_from_proper_timezone }, "id = #{self.id}")
       @destroyed = false
       reload
     end
 
+    private
+
+    def current_time_from_proper_timezone
+      ActiveRecord::Base.default_timezone == :utc ? Time.now.utc : Time.now
```

Comment:
I'm using the exact method that Rails is using to update internally the `updated_at` on doing a touch. Unluckily its `private` not `protected`. See here: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/timestamp.rb#L114